12/31 성훈 카드

BMDK9 opened the pull request.

BMDK9 committed:
feat : Card Entity 구축

BMDK9 committed:
feat : CardCreate 관련 Dto 구축

BMDK9 committed:
chore : stage getter 붙임

BMDK9 committed:
refactor : CreateCardRequestDto 필수 인자 설정

BMDK9 committed:
chore : Card Entity nullable

BMDK9 committed:
feat : Card Constant 구축

BMDK9 committed:
refactor : CreateCardResponseDto 매개변수 수정

BMDK9 committed:
feat : MVC 구축 및 CreatCard 1차 구현

BMDK9 committed:
refactor : CreateCard 코드 수정

BMDK9 committed:
feat : UpdateCardDto 구축

BMDK9 committed:
refactor : card entity 수정

BMDK9 committed:
feat : updateCardStage 기능 구현

BMDK9 committed:
feat : updateCardStage 기능 구현

BMDK9 committed:
feat : updateCardTitle 기능 구현

BMDK9 committed:
chore : 필드 이름 변경

BMDK9 committed:
feat : updateCardAccount 기능 구현

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/card/controller/CardController.java:
```diff
@@ -0,0 +1,72 @@
+    private final CardMatcherService cardMatcherService;
+
+    @PostMapping
+    public CreateCardResponseDto createCard(@PathVariable Long board_id,
```
ResponseEntity<CreateCardResponseDto> 로 변경해주실 수 있나요~?

(다른 부분도,,)

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/card/controller/CardController.java:
```diff
@@ -0,0 +1,72 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/boards/{board_id}/stage/{stage_id}/cards")
```
- http header값이나 uri 값에 대한 검증 :: Controller 단에서 검증 => 다른 도메인 검증 시, Controller가 다른 도메인 서비스 의존하게 변경

- http body값이나 uri 값이 로직이랑 밀접한 검증 :: Service 단에서 검증

1. 검증을 분리의 이점 :: 위치에 따라 검증을 분리해줄 수 있음
2. 리팩토링이 하기 편하지 않나

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/core/service/CardMatcherService.java:
```diff
@@ -0,0 +1,97 @@
+    private final CardService cardService;
+
+
+    public CreateCardResponseDto createCard(Long boardId, Long stageId, User user,
+            CreateCardRequestDto requestDto) {
+
```
throws 문을 사용해서 해당 메서드가 어떤 Exception들을 던질 수 있는지 명시해주실 수 있나요??

vanillacake369 replied:
`public void test(String a, String b) throws NumberFormatException,DomainRunTimeException ,,,`

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/card/service/CardService.java:
```diff
@@ -0,0 +1,82 @@
+        return CreateCardResponseDto.of(saveCard, stage, assignor);
+    }
+
+    public UpdateCardStageResponseDto updateStage(Stage afterStage, Card card) {
+        card.updateStage(afterStage);
+        return UpdateCardStageResponseDto.of(card);
+    }
+
+    public UpdateCardTitleResponseDto updateTitle(String title, Card card) {
+        card.updateTitle(title);
+        return UpdateCardTitleResponseDto.of(card);
+    }
+
+    public UpdateCardAccountResponseDto updateAccount(User assignor, Card card) {
+        card.updateAssignorAccount(assignor);
+        return UpdateCardAccountResponseDto.of(card);
+    }
+    private Card checkCard(Long cardId) {
+        return cardRepository.findById(cardId).orElseThrow(
+                ()-> new NullPointerException("없는 카드입니다."));
+    }
+
+    public Card checkStageCard(Long stageId, Long cardId) {
+        Card card = checkCard(cardId);
+        if (!card.getStage().getId().equals(stageId)) {
+            throw new IllegalArgumentException("해당 카드는 다른 스테이지에 있습니다.");
+        }
+        return card;
+    }
+
+
+    public void checkAssignor(Long cardId, User user) {
+        Card card = checkCard(cardId);
+        if (!card.getAssignor().getId().equals(user.getId())) {
+            throw new IllegalArgumentException("해당 권한이 없습니다");
+        }
+    }
```
`@Transactional` 빼먹으신 것 같아요!

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/card/entity/Card.java:
```diff
@@ -1,58 +1,91 @@
 
-	@OneToMany(mappedBy = "card", targetEntity = UserCard.class, cascade = CascadeType.ALL, orphanRemoval = true)
-	private final List<UserCard> userCards = new ArrayList<>();
+    @Column
+    private Boolean isDone;
```
isDone 삭제 해주세용~~

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/card/entity/Card.java:
```diff
@@ -1,58 +1,91 @@
+    @Column(nullable = false)
+    private String title;
+
+    @Column
+    private Integer priority;
```
Enum 값을 통해서 높음,중간,낮음,없음으로 @EnumeratedValue 값을 지정해주실 수 있나요~?

vanillacake369 commented on src/main/java/com/sparta/givemetuna/domain/card/dto/request/UpdateCardAccountRequestDto.java:
```diff
@@ -0,0 +1,9 @@
+@Getter
+public class UpdateCardAccountRequestDto {
+
+    private String assignorAccount;
```
@NotNull
@Size(min = 4, max = 15)
@NotBlank

이런 식의 형식값에 대해 지정해주셔서 컨트롤러 단에서 `@Valid` 설정해주시면 감사할 것 같아요~~~!!

vanillacake369 reviewed (changes requested):
고생하셨습니당~! 해당 부분만 살펴봐주세요~~

BMDK9 committed:
refactor : 1차 수정

1.ResponseEntity 로 감싸기 완료
2.Controller 에서 uri 검증 완료
3.throws 로 Exception 던지기 X(=미구현)
4.@Transactional 수정
5.isDone 삭제 완료
6.Priority Enum 구현 완료
7.controller validation 수정 완료

BMDK9 committed:
clean : 코드 다듬기

BMDK9 committed:
feat : updateCardPriority 구현

BMDK9 committed:
feat : updateCardPeriod 구현(카드 기한 설정)

BMDK9 committed:
feat : getCard 구현

BMDK9 committed:
feat : getCardList 구현

BMDK9 committed:
feat : getCardList 페이징 처리 구현

BMDK9 committed:
feat : cardDelete 구현

BMDK9 committed:
refactor : service 구조 개편

BMDK9 committed:
chore : 자잘한 수정

BMDK9 committed:
refactor : card 페이징 반환을 Page 로

BMDK9 committed:
chore : @Transactional 수정, 코드 정리

lsj135779 committed:
Merge branch 'develop' into feature/castleHun
